src/mongo/db/dbcommands_generic.cpp

```diff
@@ -275,12 +275,18 @@ class ListCommandsCmd : public Command {
                      int,
                      string& errmsg,
                      BSONObjBuilder& result) {
+        // sort the commands before building the result BSON
+        std::map<string, Command*> sortedCommands;
+        for (CommandMap::const_iterator it = _commands->begin(); it != _commands->end(); ++it) {
+            sortedCommands.emplace(it->first, it->second);
+        }
```

Comment:
std::map<K, V> and std::unordered_map<K, V> have the same value_type typedef. I wonder if you could just write this as:

```
std::map<CommandMap::key_type, CommandMap::value_type> sortedCommands(
    _commands->begin(), _commands->end());
```
Also, std::map here feels like overkill. Commands number in the hundreds, sorting a vector of that size is going to be very fast.
```
std::vector<CommandMap::value_type> commands(_commands.begin(), _commands.end());
std::sort(commands.begin(), commands.end(), []...);
```
I also notice that in the loop below you only use the first element of the pair once, to filter. In that case, do the filtering when adding to the vector, and only keep the command pointer, avoiding the string copies, then sort that with a lambda based on the name in the command.
```
std::vector<Command*> commands;
for (const auto& entry : _commands)
    if (entry.first == entry.second->name)
        commands.push_back(entry.second);
std::sort(commands.begin(), commands.end(), [](Command* lhs, Command* rhs) {
    return (lhs->name) < (rhs->name);
});
```